### PR TITLE
Filter out duplicates when showing lightbulb results.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
As an example, here was hte list before:

image

Note the multiple entries for "System.Collections.Generic.IList" and "CodeAnalysis.CodeActions.CodeAction", as well as "GenerateType."

This is because the way the margin-lightbulb works is that it collects all the fixes for the line and displays them.  In general, this isn't a bad idea (especially as the lightbulb is on the margin, and so is really associated with the line).  However, in some cases it can lead to a very subpar experience.  This is especially the case when a single line has multiple copies of the same issue.  For example a line that references the same unbound type multiple times.

I've added a few heuristics to clean things up.  First off, any light bulb actions that intersect the caret span are always displayed.  This way if you click on an issue and then bring up the lightbulb, you always see all the fixes relevant to that span.  However, after including all those actions, we only include the remaining actions on the line *if* they are not a duplicate of an action we've already added.  This way we don't add more clutter to hte list (which can just make things look much more confusing).

After this change the list looks like:

image

**Comment by CyrusNajmabadi** [comment]
Tagging @dotnet/roslyn-ide

**Comment by CyrusNajmabadi** [comment]
tagging @DustinCampbell @kuhlenh

**Comment by CyrusNajmabadi** [comment]
Note: an issue i need to look into is why "Spell check 'CodeAction'" seems to appear in the group related to "IList".  It would be a lot more sensible in the group with all the CodeAction fixes.

**Comment by CyrusNajmabadi** [comment]
Fix for spell-check item being in the wrong place is here: https://github.com/dotnet/roslyn/pull/14448

**Comment by CyrusNajmabadi** [comment]
With that fix, the list now looks like: 

image

**Comment by DustinCampbell** [comment]
Screenshots are looking great! :smile:

File: src/Features/Core/Portable/CodeFixes/Qualify/AbstractFullyQualifyCodeFixProvider.cs [hunk 1]
```diff
@@ -120,6 +119,15 @@ protected AbstractFullyQualifyCodeFixProvider()
             }
         }
 
+        private static string GetNodeName(Document document, SyntaxNode node)
+        {
+            var syntaxFacts = document.Project.LanguageServices.GetService<ISyntaxFactsService>();
+            string name;
+            int arity;
+            syntaxFacts.GetNameAndArityOfSimpleName(node, out name, out arity);
```
Comment (DustinCampbell) [inline src/Features/Core/Portable/CodeFixes/Qualify/AbstractFullyQualifyCodeFixProvider.cs]: I can't wait to be able to use C# 7 here.

**Comment by CyrusNajmabadi** [comment]
retest windows_release_unit32_prtest  please

**Pull request merged**